hilti/toolchain/src/compiler/codegen/codegen.cc

```diff
@@ -279,6 +279,19 @@ struct Visitor : hilti::visitor::PreOrder {
 
         module = n->id();
         unit->addInitialization(cg->compile(n->statements()));
+
+        // Add imported types.
+        for ( auto* import : n->childrenOfType<declaration::ImportedModule>() ) {
+            auto uid = import->uid();
+            assert(uid);
+
+            auto* m = cg->context()->astContext()->module(*uid);
+
+            for ( auto* decl : m->declarations() ) {
+                if ( decl->isA<declaration::Type>() )
+                    n->add(cg->context()->astContext(), decl);
```

Comment:
With the changes we now emit many more type infos into C++ code (we emit a type info for each type _declared_ in a module). A naive fix would seem to be to add a check for same module so we only emit type infos into C++ files for modules declaring the type, https://github.com/zeek/spicy/blob/4a1b43ef07d1305a7e88a4f0866068dc49de9d06/hilti/toolchain/src/compiler/codegen/types.cc#L1198

This fails however since with that we have no C++ file to emit type infos of builtin types from e.g., `spicy_rt` or `hilti` to. For this to work we would either add artificial units for builtin libraries, or maybe even better build them into `libspicy` and `libhilti`.